development/src/logic/pieces.ts

```diff
@@ -78,8 +78,36 @@ export type Square = {
 };
 
 export class Pawn extends Piece {
+
+  enPassant: boolean;
+
   constructor(position: Position, player: Player) {
     super(position, player, pawnResource, 'Pawn');
+    this.enPassant = false;
+  }
+
+  enPassantCheck(targetPosition: Position, draggedPiece: Piece){
+
+    let changeInPosition = 0;
+  
+    if (targetPosition.coordinates[1] === 2) {
+      changeInPosition = 1;
+    }
+    else if (targetPosition.coordinates[1] === 5){
+      changeInPosition = -1;
```

Comment:
What happens if a white Pawn goes all the way to y=2? Shouldn't this chexk also depend on the piece's color and whether or not it is a pawn?

Reply:
Nothing cause their enPassant will be equal to false so it will just cancel out. Perhaps it would be more efficient to also check the color in those if statements.

Reply:
perhaps